packages/context/src/__tests__/acceptance/binding-config.acceptance.ts

```diff
@@ -78,6 +78,47 @@ describe('Context bindings - injecting configuration for bound artifacts', () =>
     expect(server1.port).to.eql(3000);
   });
 
+  it('allows configPath for injection metadata', async () => {
+    class RestServerWithPort {
+      constructor(@config({configPath: 'port'}) public port: number) {}
+    }
+
+    // Bind configuration
+    ctx
+      .configure('servers.rest.server1')
+      .toDynamicValue(() => Promise.resolve({port: 3000}));
+
+    // Bind RestServer
+    ctx.bind('servers.rest.server1').toClass(RestServerWithPort);
+
+    // Resolve an instance of RestServer
+    // Expect server1.config to be `{port: 3000}
+    const server1 = await ctx.get<RestServerWithPort>('servers.rest.server1');
+    expect(server1.port).to.eql(3000);
+  });
+
+  it('allows targetBindingKey for injection metadata', async () => {
```

Comment:
I find the name `targetBindingKey` rather long. Is it important to include "binding" in the name? Would `targetKey` be enough to convey the meaning of this option? 

I am not sure. If you feel it's important to include "binding" in the name then I am ok with that. It's better to have a longer option name than a name that's not clear.

Reply:
How about `forBinding`?

```ts
@config({forBinding: 'restServer', configPath: 'port'})
```

Or even `for`? Would that be too cryptic?

```ts
@config({for: 'restServer', configPath: 'port'})
```

Reply:
Actually, I have a question: is `targetBindingKey` specifying the key used to resolve the config (e.g. as returned by `buildKeyForConfig`) or the key of the binding that the configuration is for? My understanding is that the second option is correct, my comments above are based on that assumption too.

Let's improve the API docs to make this fact crystal clear. I think it would be great to add a code example showing how to use custom target key.

Related question: does it make any sense to allow `@config()` to provide the custom _config binding key_ (as returned for `buildKeyForConfig`) too? Or can is this use-case something that's already solvable by using `@inject` directly? Either way, I guess this is out of scope of this pull request, but it may be worth capturing in our docs.

Reply:
I now use `fromBinding`.